### PR TITLE
Feature/call next ads on period

### PR DESCRIPTION
**Pull request by sambua**

### CODE REVIEW

File: src/dynamics/video_player/player/player.js [hunk 1]
```diff
@@ -2004,6 +2017,118 @@ Scoped.define("module:VideoPlayer.Dynamics.Player", [
...
+                },
+
+                getNextOutstreamAdTagURLs: function(immediate) {
+                    immediate = immediate || false;
+                    var promise = Promise.create();
+                    if (this.get("outstreamoptions.maxadstoshow") > -1) {
+                        if (this.get("outstreamoptions.maxadstoshow") === 0) {
+                            return promise.asyncError("Limit of showing ads exceeded.", true);
+                        }
+                    }
```
Comment (ntshcalleia) [inline src/dynamics/video_player/player/player.js:2064]:
```suggestion
                    if (this.get("outstreamoptions.maxadstoshow") === 0) {
                        return promise.asyncError("Limit of showing ads exceeded.", true);
                    }
```

File: src/dynamics/video_player/player/player.js [hunk 2]
```diff
@@ -2004,6 +2017,118 @@ Scoped.define("module:VideoPlayer.Dynamics.Player", [
...
+                    if ((this.get("nextadtagurls") && this.get("nextadtagurls").length > 0) || (this.get("adtagurlfallbacks") && this.get("adtagurlfallbacks").length > 0)) {
+                        promise.asyncSuccess(this.get("nextadtagurls").length > 0 ? this.get("nextadtagurls").shift() : this.get("adtagurlfallbacks").shift());
+                    } else {
+                        Async.eventually(function() {
+                            var isGlobalPromise = typeof this.requestForTheNextAdTagURL().then === "function";
```
Comment (ntshcalleia) [inline src/dynamics/video_player/player/player.js:2069]: @sambua this is creating an extra call for `requestForTheNextAdTagURL()`. 😬

**Review by ntshcalleia — CHANGES_REQUESTED** [review]
Hi @sambua, I left a few comments that need to be addressed. Apart from those it looks good to me. 🙂

**Comment by sambua** [comment]
Hi @ntshcalleia , thanks taking time and for detailed review. I'll make changes and re-push.

**Comment by sambua** [comment]
Hi @ntshcalleia I've made required changes, please review when you have a time 🙂 .

**Review by ntshcalleia — APPROVED** [review]

File: src/dynamics/video_player/player/states.js [hunk 3]
```diff
@@ -465,10 +479,20 @@ Scoped.define("module:VideoPlayer.Dynamics.PlayerStates.LoadAds", [
...
                     this.listenOn(this.dyn.channel("ads"), "ad-error", function() {
-                        if (this.dyn.get("adtagurlfallbacks") && this.dyn.get("adtagurlfallbacks").length > 0) {
+                        if (this.dyn.get("outstream")) {
+                            this.dyn.hidePlayerContainer();
+                            if ((this.dyn.get("nextadtagurls") && this.dyn.get("nextadtagurls").length > 0) || (this.dyn.get("adtagurlfallbacks") && this.dyn.get("adtagurlfallbacks").length > 0)) {
+                                this.dyn.set("adtagurl", this.dyn.get("nextadtagurls").length > 0 ? this.dyn.get("nextadtagurls").shift() : this.dyn.get("adtagurlfallbacks").shift());
+                                this.dyn.scopes.adsplayer.execute("requestAds");
+                            } else {
+                                this.dyn.setNextOutstreamAdTagURL(false, this);
+                            }
```
[the remark this answers is not in the document]
Comment (sambua) [reply src/dynamics/video_player/player/states.js]: hi @ntshcalleia , just remember reason why I've left this conditions. When first ad will not be started ads player will be broke. Above condition allow got url from the `nextadtagurls` defined inside `_started` method if `adshassource` not exists, so will put it back.

**Comment by ntshcalleia** [comment]
Hi @sambua, can you please remove the changes to the `dist` folder or resolve the merge conflicts?

**Comment by sambua** [comment]
Hi @ntshcalleia I thought I've already did it. Now it's done

**Pull request merged**